### PR TITLE
Optimize InstrumentFieldsMiddleware when metrics are disabled

### PR DESCRIPTION
**Pull request by sungam3r**

### CODE REVIEW

File: src/GraphQL/Instrumentation/InstrumentFieldsMiddleware.cs [hunk 1]
```diff
@@ -6,7 +6,14 @@ namespace GraphQL.Instrumentation
     public class InstrumentFieldsMiddleware : IFieldMiddleware
     {
         /// <inheritdoc/>
-        public async ValueTask<object?> ResolveAsync(IResolveFieldContext context, FieldMiddlewareDelegate next)
+        public ValueTask<object?> ResolveAsync(IResolveFieldContext context, FieldMiddlewareDelegate next)
```
Comment (sungam3r) [inline src/GraphQL/Instrumentation/InstrumentFieldsMiddleware.cs:9]: Sync when metrics are disabled.

File: src/GraphQL/Instrumentation/InstrumentFieldsMiddleware.cs [hunk 2]
```diff
@@ -6,7 +6,14 @@ namespace GraphQL.Instrumentation
...
+        {
+            return context.Metrics.Enabled
+                ? ResolveWhenMetricsEnabledAsync(context, next)
+                : default;
```
Comment (Shane32) [inline src/GraphQL/Instrumentation/InstrumentFieldsMiddleware.cs:13]:
```suggestion
                : next(context);
```
??

Comment (sungam3r) [reply src/GraphQL/Instrumentation/InstrumentFieldsMiddleware.cs:13]: oooooops

**Review by Shane32 — APPROVED** [review]

File: src/GraphQL/Instrumentation/InstrumentFieldsMiddleware.cs (hunk 2 above)
Comment (sungam3r) [reply src/GraphQL/Instrumentation/InstrumentFieldsMiddleware.cs]: I thought one thing, but wrote the other.

**Pull request merged**